### ISSUE TITLE
ProtocolWrapper should jam itself in as the transport in makeConnection, not connectionMade.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3686]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)| @mwhudson reported|
|-|-|
|Trac ID|trac#3686|
|Type|enhancement|
|Created|2009-03-11 22:50:07Z|
|Branch|https://github.com/twisted/twisted/tree/protocolwrapper-makeconnection-3686|

Arguably, anyway.

<details><summary>Searchable metadata</summary>

```
trac-id__3686 3686
type__enhancement enhancement
reporter__mwh mwh
priority__normal normal
milestone__ 
branch__branches_protocolwrapper_makeconnection_3686 branches/protocolwrapper-makeconnection-3686
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1236811807000000 1236811807000000
changetime__1237478548000000 1237478548000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

By overriding `connectionMade` instead, `ProtocolWrapper` is not precisely adhering to `IProtocol`.  It probably shouldn't have a `connectionMade` at all, just a `makeConnection` which does everything.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [26377]) Branching to 'protocolwrapper-makeconnection-3686'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [26378]) Drop ProtocolWrapper.connectionMade in favor of a more complete ProtocolWrapper.makeConnection

As a bonus, also get rid of some other cruft.

refs [#3686](https://github.com/twisted/twisted/issues/3686)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/protocolwrapper-makeconnection-3686)

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_5"></a>|@mwhudson set owner to @exarkun|
|-|-|

Looks fine.

I don't understand how test_transportInterfaces still passes, but it does, so...

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [26426]) Merge protocolwrapper-makeconnection-3686

Author: exarkun
Reviewer: mwhudson
Fixes: [#3686](https://github.com/twisted/twisted/issues/3686)

Simplify `ProtocolWrapper` by consolidating its setup code in its
`makeConnection` method.  Also remove the redundant `directlyProvides`
call it makes (redundant because `__getattr__` causes it to provide
the interfaces of the wrapped transport already).  Also improve test
coverage and remove a redundant test helper.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
